### PR TITLE
Worker installation script changes to deal with versioning

### PR DESCRIPTION
**Pull request by rapoth**
This PR brings changes to accommodate versioning in our releases.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [ ] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ ] You have included any necessary tests in the same PR.

### CODE REVIEW

File: deployment/install-worker.sh [hunk 1]
```diff
@@ -1,40 +1,69 @@
...
+# Description:
+# This is a helper script to install the worker binaries on your Apache Spark cluster
+#
+# Usage:
+# ./install-script.sh <cloud-provider> <path-to-worker> <local-worker-installation-path>
+#
+# Sample usage:
+# ./install-script.sh 
+#   azure 
+#   abfs://<blobcontainer>@<gen2storageaccount>.dfs.core.windows.net/<path>/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.1.0.tar.gz
```
Comment (imback82) [inline deployment/install-worker.sh]: I am wondering if we should move to `wget` and let the user to point to our release on GitHub (verified on HDInsight). We don't have to do this in this PR, but we may want to track this if we want to simplify the experience.
cc: @suhsteve

File: deployment/install-worker.sh [hunk 2]
```diff
@@ -1,40 +1,69 @@
...
+
 # The path on the executor nodes where Microsoft.Spark.Worker executable is installed.
-DEST_PATH=$3
+LOCAL_WORKER_PATH=$3
+
 # The path where all the dependent libraies are installed so that it doesn't
-# pollute the $DEST_PATH.
-DEST_PATH_BINARIES=$DEST_PATH/microsoft.spark.worker
+# pollute the $LOCAL_WORKER_PATH.
+SPARKDOTNET_ROOT=$LOCAL_WORKER_PATH/spark-dot-net
```
Comment (imback82) [inline deployment/install-worker.sh]: `dotnet`

File: deployment/install-worker.sh [hunk 3]
```diff
@@ -1,40 +1,69 @@
...
+#   /usr/local/bin
+##############################################################################
+
+set +e
 # Cloud Provider
 CLOUD_PROVIDER=$1
+
 # Path where packaged worker file (tgz) exists.
-WORKER_PATH=$2
+WORKER_PATH_OR_URI=$2
```
Comment (imback82) [inline deployment/install-worker.sh]: I think we need SRC_ and DEST_ . LOCAL_WORKER_PATH is still not clear.

File: deployment/install-worker.sh [hunk 4]
```diff
@@ -1,40 +1,69 @@
...
-sudo rm -rf $DEST_PATH_BINARIES
+# Extract version
+IFS='-' read -ra BASE_FILENAME <<< "$(basename $WORKER_PATH_OR_URI .tar.gz)"
+VERSION=${BASE_FILENAME[2]}
+
+IFS='.' read -ra VERSION_CHECK <<< "$VERSION"
+[[ ${#VERSION[@]} != 3 ]] || { echo >&2 "Version check does not satisfy. Raise an issue here: https://github.com/dotnet/spark"; exit 1; }
+
+# Path of the final worker binaries (the one we just downloaded and extracted)
+CURRENT_WORKER_PATH=$SPARKDOTNET_ROOT/Microsoft.Spark.Worker-$VERSION
```
Comment (imback82) [inline deployment/install-worker.sh]: why is this current and the comment is final?

Comment (rapoth) [reply deployment/install-worker.sh]: Done

File: deployment/install-worker.sh (hunk 3 above)
Comment (rapoth) [reply deployment/install-worker.sh]: Done

File: deployment/install-worker.sh (hunk 2 above)
Comment (rapoth) [reply deployment/install-worker.sh]: Done

File: deployment/install-worker.sh (hunk 1 above)
Comment (rapoth) [reply deployment/install-worker.sh]: Added the capability.

**Pull request merged**